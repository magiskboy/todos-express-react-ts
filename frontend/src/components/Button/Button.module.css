.button {
    background-color: #57C0BA;
    color: white;
    padding: 18px 8px;
    border: none;
    font-family: 'Inter';
}