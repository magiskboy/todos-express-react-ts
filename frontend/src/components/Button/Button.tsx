import React from "react";
import s from "./Button.module.css";

type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement>;

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    return (
      <button
        ref={ref}
        {...props}
        className={`${s.button} ${props.className ? props.className : ""}`}
        data-testid="button"
      >
        {props.children}
      </button>
    );
  }
);
