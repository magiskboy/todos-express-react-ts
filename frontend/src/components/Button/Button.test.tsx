import { fireEvent, render } from "@testing-library/react";
import { Button } from "./Button";

describe("Button", () => {
    it("renders button with custom class", () => {
        const { getByText } = render(
            <Button className="custom-button">Click me</Button>
        );
        const button = getByText("Click me");
        expect(button.className).toContain("custom-button");
    });

    it("calls onClick when button is clicked", () => {
        const onClickMock = jest.fn();
        const { getByText } = render(
            <Button onClick={onClickMock}>Click me</Button>
        );
        const button = getByText("Click me");
        fireEvent.click(button);
        expect(onClickMock).toHaveBeenCalled();
    });

    it("matches snapshot", () => {
        const { container } = render(<Button>Click me</Button>);
        expect(container).toMatchSnapshot();
    });
});