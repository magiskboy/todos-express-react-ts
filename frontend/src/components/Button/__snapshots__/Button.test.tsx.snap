// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div>
  <button
    class="button "
    data-testid="button"
  >
    Click me
  </button>
</div>
`;
